pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -643,12 +674,20 @@ func (facts *PipelineRunFacts) getPipelineTasksCount() pipelineRunStatusCount {
 		// increment success counter since the task is successful
 		case t.isSuccessful():
 			s.Succeeded++
+		// increment failure counter since the task is cancelled due to a timeout
+		case t.isCancelledForTimeOut():
+			s.Failed++
 		// increment cancelled counter since the task is cancelled
 		case t.isCancelled():
 			s.Cancelled++
 		// increment failure counter since the task has failed
 		case t.isFailure():
 			s.Failed++
+		// increment failure counter since the task was skipped due to the pipeline, tasks, or finally timeout being reached before the task was launched
+		case t.Skip(facts).SkippingReason == v1beta1.PipelineTimedOutSkip ||
+			t.Skip(facts).SkippingReason == v1beta1.TasksTimedOutSkip ||
+			t.IsFinallySkipped(facts).SkippingReason == v1beta1.FinallyTimedOutSkip:
+			s.Failed++
```

Comment:
The reason for this is that the current behavior when a `TaskRun` or `Run` is "created" after the appropriate timeout (`.timeouts.pipeline`, `.timeouts.tasks`, `.timeouts.finally`) has been reached is to actually create the `TaskRun`/`Run`, but with a timeout of `1s`, so that it fails immediately. While I believe that having `rpt.Skip(facts).IsSkipped` return true for `ResolvedPipelineTask`s which shouldn't be created due to timeouts is a much better approach, without this additional change, the `PipelineRun` wouldn't end up marked as failed, because the tasks are marked as skipped. So let's categorize them as failed instead here.

Reply:
What about `PipelineRun.Status.SkippedTasks` -- I think it'd be confusing if these `Tasks` are still listed in `SkippedTasks` but they are not counted as skipped Tasks in another part of the status

Reply:
That's a good question. I guess we could special-case things to keep them out of `SkippedTasks` too...

Also, in practice, there's a real question as to whether it actually matters if they're marked as cancelled/skipped or failed - if the `PipelineRun` gets past the relevant timeout, there's only the tiniest of chances that all previous tasks succeeded before it hit the timeout. So perhaps we can just not worry about it and update the tests accordingly.

Reply:
or we could leave them as skipped tasks instead of counting these skipped tasks as failed tasks in this section of the status only

Reply:
Actually, I think I like the idea of adding a `SkippedDueToTimeout` counter alongside `Skipped`, `Failed`, etc, incrementing both that counter and `Skipped` when a task is skipped due to timeout. Then, when setting the `PipelineRun` status, we change the "did this fail" case to `s.Failed > 0 || s.SkippedDueToTimeout > 0`. That keeps the overall `PipelineRun` status as failed for timeout skips, but also keeps the tasks in `SkippedTasks`.

Reply:
I like that idea, let's do it 🚀

Reply:
Done. =)